src/main/java/net/minecraftforge/common/crafting/GrindingRecipe.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.crafting;
+
+import com.google.gson.JsonObject;
+
+import net.minecraft.inventory.IInventory;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.IRecipeType;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.item.crafting.ShapedRecipe;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.registries.ForgeRegistryEntry;
+
+/**
+ * A Recipe for the grindstone. It takes 2 ingredients as input and returns an ItemStack.
+ */
+public class GrindingRecipe implements IRecipe<IInventory>
+{
+    public static final IRecipeSerializer<GrindingRecipe> SERIALIZER = new Serializer();
+    
+    private ResourceLocation id;
+    private Ingredient base;
+    private Ingredient addition;
+    private ItemStack result;
+    
+    public GrindingRecipe(ResourceLocation id,Ingredient ingredient, Ingredient ingredient1, ItemStack result) 
```

Comment:
Once again with the mismatched param and field names. Make a decision to rename one or the other.